pkgs/cli/test/integration.zig

```diff
@@ -0,0 +1,193 @@
+const std = @import("std");
+const process = std.process;
+const net = std.net;
+const build_options = @import("build_options");
+
+test "CLI beam command with mock network - complete integration test" {
+    const allocator = std.testing.allocator;
+
+    // Verify executable exists first
+    const exe_file = std.fs.openFileAbsolute(build_options.cli_exe_path, .{}) catch |err| {
+        std.debug.print("ERROR: Cannot find executable at {s}: {}\n", .{ build_options.cli_exe_path, err });
+
+        // Try to list the directory to see what's actually there
+        std.debug.print("INFO: Attempting to list zig-out/bin directory...\n", .{});
+        const dir_path = std.fs.path.dirname(build_options.cli_exe_path);
+        if (dir_path) |path| {
+            var dir = std.fs.openDirAbsolute(path, .{ .iterate = true }) catch |dir_err| {
+                std.debug.print("ERROR: Cannot open directory {s}: {}\n", .{ path, dir_err });
+                return err;
+            };
+            defer dir.close();
+
+            var iterator = dir.iterate();
+            std.debug.print("INFO: Contents of {s}:\n", .{path});
+            while (try iterator.next()) |entry| {
+                std.debug.print("  - {s} (type: {})\n", .{ entry.name, entry.kind });
+            }
+        }
+
+        return err;
+    };
+    exe_file.close();
+    std.debug.print("INFO: Found executable at {s}\n", .{build_options.cli_exe_path});
+
+    // Start CLI with beam command and mock network - use build option for executable path
+    const args = [_][]const u8{ build_options.cli_exe_path, "beam", "--mockNetwork", "true" };
+    var cli_process = process.Child.init(&args, allocator);
+
+    // Capture stdout and stderr for debugging
+    cli_process.stdout_behavior = .Pipe;
+    cli_process.stderr_behavior = .Pipe;
+
+    defer {
+        _ = cli_process.kill() catch {};
+        _ = cli_process.wait() catch {};
+    }
+
+    // Start the process
+    cli_process.spawn() catch |err| {
+        std.debug.print("ERROR: Failed to spawn process: {}\n", .{err});
+        return err;
+    };
+
+    std.debug.print("INFO: Process spawned successfully with PID\n", .{});
+
+    // Wait for metrics server to be ready (with extended timeout for CI)
+    const metrics_port: u16 = 9667;
```

Comment:
put all these constants in cli/constants file ad DEFAULT_METRICS_PORT (for e.g.) and import it over here to whatever the defaults are